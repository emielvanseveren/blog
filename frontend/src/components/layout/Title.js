import React from 'react'
import PropTypes from 'prop-types'
import size from '../../constants/size'
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import Info from './Info'

const Container = styled.div`
  margin: 75px 0;
  width: 100%;
  transition: .15s transform ease-in-out;
  &:hover{
    transform: translateY(-3px);
  }
  @media ${size.xs}{
    margin: 50px auto 50px auto;
    width: 95%;
  }
`

const StyledLink = styled(Link)`
  font-size: 1.8rem;
  color: ${({ theme }) => theme.highlight};
  font-weight: 700;
  @media ${size.sm}{
    font-size: 1.5rem;
  }
  @media ${size.xs}{
    font-size: 1.3rem;
  }
`
const Description = styled.p`
  display: block;
  font-size: 0.9rem;

  @media ${size.xs}{
    font-size: 0.8rem;
  }
`

export default function Title({ date, description, read, title, postId }){

  //put postId in localstorage
  localStorage.setItem('post', JSON.stringify({ postId }))


  return (
    <Container>
      <StyledLink to={{ pathname: '/' + title.replace(/\s+/g, '-').toLowerCase()}}>{title}</StyledLink>
      <Info date={date} read={read}/>
      <Description>{description}</Description>
    </Container>
  )
}
Title.propTypes = {
  date:         PropTypes.instanceOf(Date),
  description:  PropTypes.string.isRequired,
  postId:       PropTypes.number.isRequired,
  read:         PropTypes.number.isRequired,
  title:        PropTypes.string.isRequired
}
