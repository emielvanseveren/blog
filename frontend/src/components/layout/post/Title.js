import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Container = styled.div`
  width: 600px;
  margin: 0 auto 50px auto;
`

const StyledTitle = styled.h2`
  margin-top: 50px;
  font-size: 2.4rem;
  color: ${({ theme }) => theme.text};
  font-weight: 700;
  text-align: ${({ center }) => center ? 'center' : 'left'};
  text-transform: capitalize;
`
const Description = styled.div`
  text-align: center;
  font-size: 0.8rem;
  color: ${({ theme }) => theme.highlight};
  font-weight: 600;
`
export default function Title({ center = false, text, date, readTime }){
  return (
    <Container>
      <StyledTitle center={center}>{ text }</StyledTitle>
      <Description>{ date } â€¢ { readTime }</Description>
    </Container>
  )
}

Title.propTypes = {
  center:   PropTypes.bool,
  date:     PropTypes.string.isRequired,
  readTime: PropTypes.string.isRequired,
  text:     PropTypes.string.isRequired
}
