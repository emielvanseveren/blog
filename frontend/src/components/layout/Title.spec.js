import React from 'react'
import Title from './Info'

describe('<Title/>', () => {
  it('Snapshot', () => {
    expect(mountWithRouterAndTheme(
      createComponent()
    )).toMatchSnapshot()
  })
})

function createComponent(props){
  const defaultProps = {
    date: new Date('10/10/2019'),
    description: 'this is a description',
    postId: 1,
    title: 'this is a title',
    read: 5
  }
  return <Title { ...props } { ...defaultProps}/>
}
