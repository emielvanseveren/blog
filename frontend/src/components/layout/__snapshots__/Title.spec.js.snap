// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> Snapshot 1`] = `
<Info
  date={2019-10-10T00:00:00.000Z}
  description="this is a description"
  postId={1}
  read={5}
  title="this is a title"
>
  <Info__Container>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Info__Container-sc-2rdfjf-0",
            "isStatic": false,
            "lastClassName": "dBrlcP",
            "rules": Array [
              "display:block;font-size:0.7rem;margin-bottom:5px;&.post-info{@media ",
              "(max-width: 768px)",
              "{p{padding-left:10px;}}}",
            ],
          },
          "displayName": "Info__Container",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Info__Container-sc-2rdfjf-0",
          "target": "div",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="Info__Container-sc-2rdfjf-0 dBrlcP"
      >
        <p>
          November 10, 2019
           â€¢ 
          5
           min read 
          ðŸ”¥ðŸ”¥
        </p>
      </div>
    </StyledComponent>
  </Info__Container>
</Info>
`;
